presto-plan-checker-router-plugin/src/test/java/com/facebook/presto/router/scheduler/TestPlanCheckerRouterPlugin.java

```diff
@@ -0,0 +1,175 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.scheduler;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.airlift.log.Logging;
+import com.facebook.presto.client.QueryError;
+import com.facebook.presto.client.QueryResults;
+import com.facebook.presto.common.ErrorType;
+import com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils;
+import com.facebook.presto.server.MockHttpServletRequest;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.router.RouterRequestInfo;
+import com.facebook.presto.spi.router.Scheduler;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ListMultimap;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_CATALOG;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_SCHEMA;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_TIME_ZONE;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_USER;
+import static com.facebook.presto.common.ErrorType.USER_ERROR;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createRegion;
+import static com.facebook.presto.sidecar.NativeSidecarPluginQueryRunnerUtils.setupNativeSidecarPlugin;
+import static java.util.Collections.emptyList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertSame;
+import static org.testng.Assert.assertTrue;
+
+public class TestPlanCheckerRouterPlugin
+        extends AbstractTestQueryFramework
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+    private PlanCheckerRouterPluginConfig planCheckerRouterConfig;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        Logging.initialize();
+
+        URI javaRouterUri = new URI("192.168.0.1");
+        URI nativeRouterUri = new URI("192.168.0.2");
+
+        URI planCheckerClusters = ((DistributedQueryRunner) getQueryRunner()).getCoordinator().getBaseUrl();
+        planCheckerRouterConfig =
+                new PlanCheckerRouterPluginConfig()
+                        .setJavaRouterURI(javaRouterUri)
+                        .setNativeRouterURI(nativeRouterUri)
+                        .setPlanCheckClustersURIs(planCheckerClusters.toString());
+    }
+
+    @Override
+    protected void createTables()
```

Comment:
I'm missing something I think - are we actually querying these tables ? The test actually only check that the destination the router plugin resolves to..